odl/tomo/operators/ray_trafo.py

```diff
@@ -31,180 +31,227 @@
+            used.
+
         interp : {'nearest', 'linear'}, optional
-            Interpolation type for the discretization of the operator
-            range.
-            Default: 'nearest'
-        discr_range : `DiscreteLp`, optional
-            Discretized space, the range of the forward projector.
-            Default: Infered from parameters.
+            Interpolation type for the discretization of the projection
+            space. This has no effect if ``proj_space`` is given explicitly.
+            Default: ``'nearest'``
+        proj_space : `DiscreteLp`, optional
+            Discretized projection (sinogram) space, the range of the forward
+            operator or the domain of the adjoint (back-projection).
+            Default: Inferred from parameters.
         use_cache : bool, optional
-            If ``True``, data is cached. Note that this causes notable memory
-            overhead, both on the GPU and on the CPU since a full volume and
-            projection is stored. In the 3D case, some users may want to
-            disable this.
+            If ``True``, data is cached. This gives a significant speed-up
+            at the expense of a notable memory overhead, both on the GPU
+            and on the CPU, since a full volume and a projection dataset
+            are stored. That may be prohibitive in 3D.
             Default: True
 
         Notes
         -----
-        The ASTRA backend is faster if data is given with ``dtype`` 'float32'
-        and storage order 'C'. Otherwise copies will be needed.
+        The ASTRA backend is faster if data are given with
+        ``dtype='float32'`` and storage order 'C'. Otherwise copies will be
+        needed.
         """
-        if not isinstance(discr_domain, DiscreteLp):
-            raise TypeError('`discr_domain` {!r} is not a `DiscreteLp`'
-                            ' instance'.format(discr_domain))
+        variant, variant_in = str(variant).lower(), variant
+        if variant not in ('forward', 'backward'):
+            raise ValueError('`variant` {!r} not understood'
+                             ''.format(variant_in))
+
+        if variant == 'forward':
+            reco_name = 'domain'
+            proj_name = 'range'
+        else:
+            reco_name = 'range'
+            proj_name = 'domain'
+
+        if not isinstance(reco_space, DiscreteLp):
+            raise TypeError('`{}` must be a `DiscreteLp` instance, got '
+                            '{!r}'.format(reco_name, reco_space))
 
         if not isinstance(geometry, Geometry):
-            raise TypeError('`geometry` {!r} is not a `Geometry` instance'
-                            ''.format(geometry))
-
+            raise TypeError('`geometry` must be a `Geometry` instance, got '
+                            '{!r}'.format(geometry))
+
+        # Handle backend choice
+        if not any(globals()[imp.upper() + '_AVAILABLE']
```

Comment:
do we truly need this implementation? If we cant do anything else, then fine, but I'd rather not.